jdaviz/core/freezable_state.py

```diff
@@ -97,15 +179,14 @@ def reset_limits(self, *event):
                 if not layer.visible or layer.layer.data.ndim == 1:
                     continue
                 pixel_ids = layer.layer.pixel_component_ids
+                pixel_id_x = [comp for comp in pixel_ids if comp.label.endswith('[x]')][0]
+                pixel_id_y = [comp for comp in pixel_ids if comp.label.endswith('[y]')][0]
 
-                x_max = max(x_max, layer.layer.data[pixel_ids[1]].max() + 0.5)
-                y_max = max(y_max, layer.layer.data[pixel_ids[0]].max() + 0.5)
+                x_max = max(x_max, layer.layer.data[pixel_id_x].max() + 0.5)
+                y_max = max(y_max, layer.layer.data[pixel_id_y].max() + 0.5)
```

Comment:
this technically isn't necessary, but will help future-proof if we ever do enable this custom `reset_limits` for any cube viewer.

Reply:
But now it depends on the label name. Is the label always guaranteed to have x/y in the name?

Reply:
that seems to be more general/reliable than assuming the first and second index 🤷‍♂️   I can revert for now though and deal with this if/when we try to make the image viewers in configs more consistent.